planner/core/plan_to_pb.go

```diff
@@ -130,8 +130,15 @@ func (p *PhysicalProjection) ToPB(ctx sessionctx.Context, storeType kv.StoreType
 			return nil, errors.Trace(err)
 		}
 		executorID = p.ExplainID().String()
+	} else if storeType == kv.TiKV {
+		var err error
+		projExec.Child, err = p.children[0].ToPB(ctx, storeType)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		executorID = p.ExplainID().String()
 	} else {
-		return nil, errors.Errorf("The projection can only be pushed down to TiFlash now, not %s.", storeType.Name())
+		return nil, errors.Errorf("The projection can only be pushed down to TiFlash or TiKV now, not %s.", storeType.Name())
```

Comment:
Unknow engine error instead?

Reply:
although we only have 2 engines supported for now, it is not exactly unknown engine, suppose we add new engines in the future, it should be the new engine does not support projection pushdown, so I prefer to keep it this way.